qiskit_optimization/algorithms/warm_start_qaoa_optimizer.py

```diff
@@ -0,0 +1,339 @@
+
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.algorithms import QAOA
+from qiskit.circuit import Parameter
+
+from qiskit_optimization import QiskitOptimizationError, QuadraticProgram
+from qiskit_optimization.algorithms import MinimumEigenOptimizationResult, MinimumEigenOptimizer, \
+    OptimizationAlgorithm, OptimizationResultStatus, SolutionSample
+from qiskit_optimization.converters import QuadraticProgramConverter
+from qiskit_optimization.problems import VarType
+
+
+class BaseAggregator(ABC):
+    """A base abstract class for aggregates results"""
+
+    def aggregate(self, results: List[MinimumEigenOptimizationResult]) \
+            -> List[SolutionSample]:
+        """
+        Aggregates the results.
+
+        Args:
+            results: List of result objects that need to be combined.
+
+        Returns:
+             Aggregated samples.
+        """
+        raise NotImplementedError
+
+
+class MeanAggregator(BaseAggregator):
+    """Aggregates the results by averaging the probability of each sample."""
+
+    def aggregate(self, results: List[MinimumEigenOptimizationResult]) \
+            -> List[SolutionSample]:
+        """
+        Args:
+            results: List of result objects that need to be combined.
+
+        Returns:
+             Aggregated samples by averaging them.
+        """
+
+        # Use a dict for fast solution look-up
+        # Key: sample code, value: tuple of fval, probability
+        dict_samples: Dict[str, Tuple[float, float]] = {}
+
+        def _to_string(x: np.ndarray) -> str:
+            return "".join(str(int(v)) for v in x)
+
+        def _from_string(string) -> np.ndarray:
+            return np.array([float(c) for c in string])
+
+        # Sum up all the probabilities in the results
+        for result in results:
+            for sample in result.samples:
+                # state, fval, prob = sample[0], sample[1], sample[2]
+                state, fval, prob = _to_string(sample.x), sample.fval, sample.probability
+                if state in dict_samples:
+                    dict_samples[state] = (fval, dict_samples[state][1] + prob)
+                else:
+                    dict_samples[state] = (fval, prob)
+
+        # Divide by the number of results to normalize
+        aggregated_samples = []
+        num_results = len(results)
+        for state in dict_samples:
+            # sample = (state, dict_samples[state][0], dict_samples[state][1] / num_results)
+            sample = SolutionSample(x=_from_string(state),
+                                    fval=dict_samples[state][0],
+                                    probability=dict_samples[state][1] / num_results,
+                                    status=OptimizationResultStatus.SUCCESS)
+            aggregated_samples.append(sample)
+
+        # new_samples.sort(key=lambda sample: problem.objective.sense.value * sample[1])
+        # x = [float(e) for e in new_samples[0][0]]
```

Comment:
Should we just get rid of this code if its not needed

Reply:
Removed.